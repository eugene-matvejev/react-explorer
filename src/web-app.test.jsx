import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WebApp from './web-app';

configure({ adapter: new Adapter() });

describe('WebApp', () => {
    const props = {
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<WebApp {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['className', '{{className}}'],
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<WebApp {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});
