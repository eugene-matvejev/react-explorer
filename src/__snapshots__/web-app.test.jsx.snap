// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WebApp/> render with default/required props 1`] = `
<section
  className="webapp "
/>
`;

exports[`<WebApp/> render with optional props [::className] as "{{className}}" 1`] = `
<section
  className="webapp {{className}}"
/>
`;
